WooCommerce/Classes/ViewRelated/Blaze/AdDestination/BlazeAdDestinationSettingViewModel.swift

```diff
@@ -27,25 +30,53 @@ final class BlazeAdDestinationSettingViewModel: ObservableObject {
         return String(format: Localization.finalDestination, buildFinalDestinationURL())
     }
 
+    // View model for the add parameter view.
+    var blazeAddParameterViewModel: BlazeAddParameterViewModel {
+
+        // The remaining characters to be used for validation in `BlazeAddParameterViewModel.
+        // If more than 1 parameter exist and user is adding a new character, the value should be reduced further by 1.
+        // This to take into account the need to add "&" separator character to the new parameter.
+        var adjustedRemainingCharacters = calculateRemainingCharacters()
+        if parameters.count > 1 && selectedParameter == nil {
+            adjustedRemainingCharacters = calculateRemainingCharacters() - 1
+        }
+
+        return BlazeAddParameterViewModel(
+            remainingCharacters: adjustedRemainingCharacters,
+            parameter: selectedParameter,
+            onCancel: { [weak self] in
+                guard let self = self else { return }
+                self.clearSelectedParameter()
+            },
+            onCompletion: { [weak self] key, value in
+                guard let self = self else { return }
+
+                if selectedParameter != nil {
+                    updateSelectedParameter(newKey: key, newValue: value)
+
+                    // Once a parameter is updated, clear the selected parameter to prepare for the next add/update action.
+                    clearSelectedParameter()
```

Comment:
Please use explicit self for now, according to this discussion p1704957841130319-slack-C6H8C3G23.